test/composition.cancel.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const metasync = require('..');
+const metatests = require('metatests');
+
+
+// Emulate Asynchronous calls of function
+//   callback - function
+// Result: undefined; call of function
+const wrapAsync = callback => {
+  setTimeout(callback, Math.floor(Math.random() * 500));
+};
+
+metatests.test('async functions composition cancel in the middle', test => {
+
+  let fc = null;
+  const fn1 = (data, cb) => {
+    wrapAsync(() => {
+      cb(null, 1);
+    });
+  };
+
+  const fn2 = (data, cb) => {
+    test.strictSame(data, [1]);
+    wrapAsync(() => {
+      if (fc) fc.cancel();
+      cb(null, 2);
+    });
+  };
+
+  const fn3 = () => {
```

Comment:
This can be replaced with `test.mustNotCall(() => {})`
or, after https://github.com/metarhia/metatests/pull/119 with `test.mustNotCall()`.